### PR TITLE
Fixing the issue with duplicate transaction synchronization processes

### PR DESCRIPTION
**Pull request by xgreenx**
Ignore peer if we already are syncing transactions from it.
Added metrics for sync and async processor.
Added metrics for all GraphQL queries.

## Checklist
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: crates/metrics/src/futures/future_tracker.rs [hunk 1]
```diff
@@ -13,11 +14,27 @@ use std::time::Instant;
...
-    pub idle: Duration,
+    idle: Duration,
     /// The output of the future.
-    pub output: Output,
+    output: Output,
+}
+
+impl<Output> ExecutionTime<Output> {
+    /// Unwraps the future output and returns the execution report.
+    pub fn unwrap(self, metric: &FuturesMetrics) -> Output {
```
Comment (MitchTurner) [inline crates/metrics/src/futures/future_tracker.rs:26]: Can we name this something else? Like `record_to_metrics` or something like that. 

The `unwrap` actually threw me off lol. I was reading it as `unwrap_or` for some `Result` and that made me confused about all the types.

**Review by AurelienFT — DISMISSED** [review]

**Comment by Amirbakhtiaro** [comment]
> Ignore peer if we already are syncing transactions from it. Added metrics for sync and async processor. Added metrics for all GraphQL queries.
> 
> ## Checklist
> * [x]  New behavior is reflected in tests
> 
> ### Before requesting review
> * [x]  I have reviewed the code myself

hotfix/duplicate-tx-syncing

**Review by AurelienFT — APPROVED** [review]

**Review by AurelienFT — DISMISSED** [review]

**Review by AurelienFT — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,9 +6,19 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
+- [2324](https://github.com/FuelLabs/fuel-core/pull/2324): Added metrics for sync, async processor and for all GraphQL queries.
+
+## Fixed
 - [2320](https://github.com/FuelLabs/fuel-core/issues/2320): Prevent `/health` and `/v1/health` from being throttled by the concurrency limiter.
 - [2322](https://github.com/FuelLabs/fuel-core/issues/2322): Set the salt of genesis contracts to zero on execution.
+- [2324](https://github.com/FuelLabs/fuel-core/pull/2324): Ignore peer if we already are syncing transactions from it.
+
+### Changed
+
+#### Breaking
```
Comment (rymnc) [inline CHANGELOG.md:19]: all of these entries(added/fixed/changes/breaking) have different sizes

is that intentional? 🤔

**Review by rymnc — COMMENTED** [review]
since this pr deals with metrics it would be nice to try using the image on devnet to ensure that the values are as intended

File: CHANGELOG.md (hunk 2 above)
Comment (xgreenx) [reply CHANGELOG.md:19]: It is fiiiiiiine)

**Review by rymnc — DISMISSED** [review]

**Review by rymnc — APPROVED** [review]
approving to not block on the changelog

File: crates/metrics/src/futures/future_tracker.rs [hunk 3]
```diff
@@ -13,11 +14,27 @@ use std::time::Instant;
...
+    idle: Duration,
     /// The output of the future.
-    pub output: Output,
+    output: Output,
+}
+
+impl<Output> ExecutionTime<Output> {
+    /// Extracts the future output and records the execution report into the metrics.
+    pub fn extract(self, metric: &FuturesMetrics) -> Output {
+        // TODO: Use `u128` when `AtomicU128` is stable.
```
Comment (Dentosal) [inline crates/metrics/src/futures/future_tracker.rs:27]: nit: we shoould probably have an issue for this

Comment (Dentosal) [reply crates/metrics/src/futures/future_tracker.rs:27]: Also it might be wise to use something like https://docs.rs/portable-atomic for now

**Review by Dentosal — APPROVED** [review]

**Pull request merged**